.github/workflows/clients-go.yml

```diff
@@ -20,45 +20,9 @@ jobs:
           name: go-binaries-${{ github.sha }}
           path: src/clients/go/pkg/native
 
-  tests:
```

Comment:
Tests are now run as part of the docs validation step.

Proven that errors in tests fail the build here: https://github.com/tigerbeetledb/tigerbeetle/pull/754